src/libbson/tests/test-string.c

```diff
@@ -356,9 +357,73 @@ test_bson_string_capacity (void *unused)
       large_str[UINT32_MAX - 2u] = 's'; // Restore.
    }
 
+   // Test allocating the largest possible string.
+   {
+      bson_string_t *str = _bson_string_alloc (2147483648);
```

Comment:
```suggestion
      bson_string_t *str = _bson_string_alloc (UINT32_MAX - 1u);
```
Reduce magic numbers.